owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/test/java/fr/openwide/core/test/wicket/more/model/TestRangeRenderer.java

```diff
@@ -0,0 +1,165 @@
+package fr.openwide.core.test.wicket.more.model;
+
+import java.util.Date;
+import java.util.Locale;
+
+import org.apache.wicket.Localizer;
+import org.apache.wicket.model.IModel;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import fr.openwide.core.test.wicket.more.AbstractWicketMoreJpaTestCase;
+import fr.openwide.core.wicket.more.model.RangeModel;
+import fr.openwide.core.wicket.more.rendering.Renderer;
+import fr.openwide.core.wicket.more.util.DatePattern;
+import fr.openwide.core.wicket.more.util.model.Models;
+
+public class TestRangeRenderer extends AbstractWicketMoreJpaTestCase {
+	
+	private static final Logger LOGGER = LoggerFactory.getLogger(TestRangeRenderer.class);
+
+	private static final String LOWER_TAG = "${start}";
+	private static final String UPPER_TAG = "${end}";
+	private static final String LOWER_UNIT_TAG = "${startUnit}";
+	private static final String UPPER_UNIT_TAG = "${endUnit}";
+	
+	private static final String KEY_DATES_BASE = "test.wicket.more.model.renderer.range.dates";
+	private static final String KEY_DATES_SOLO = KEY_DATES_BASE + ".solo";
+	private static final String KEY_DATES_BOTH = KEY_DATES_BASE + ".both";
+	private static final String KEY_DATES_FROM = KEY_DATES_BASE + ".from";
+	private static final String KEY_DATES_UPTO = KEY_DATES_BASE + ".upto";
+	
+	private static final String KEY_MONEY_BASE = "test.wicket.more.model.renderer.range.money";
+	private static final String KEY_MONEY_SOLO = KEY_MONEY_BASE + ".solo";
+	private static final String KEY_MONEY_BOTH = KEY_MONEY_BASE + ".both";
+	private static final String KEY_MONEY_FROM = KEY_MONEY_BASE + ".from";
+	private static final String KEY_MONEY_UPTO = KEY_MONEY_BASE + ".upto";
+	
+	private static final String KEY_COUNT_BASE = "test.wicket.more.model.renderer.count.money";
+
+	/**
+	 * <p>JUnit non-regression tests on Range renderers.
+	 * <p><b>Tests:</b><ul>
+	 * <li>Whether the base of the resource key is correctly used
+	 * <li>Whether the right resource key suffix is used, regarding the bounds of the given range
+	 * <li>Whether the four tags which may be found within a string designated by a resource key are correctly replaced
+	 * </ul>
+	 */
+	@Test
+	public void testRangeRenderer() {
+		testWithUnits();
+		testWithoutUnits();
+	}
+	
+	private void testWithUnits() {
+		String expected;
+		IModel<Integer> lowerMoneyModel;
+		IModel<Integer> upperMoneyModel;
+		RangeModel<Integer> moneyRangeModel;
+		// Testing .solo suffix
+		lowerMoneyModel = Models.transientModel(42);
+		upperMoneyModel = Models.transientModel(42);
+		moneyRangeModel = new RangeModel<Integer>(lowerMoneyModel, upperMoneyModel);
+		expected = Localizer.get().getString(KEY_MONEY_SOLO, null)
+				.replace(LOWER_UNIT_TAG, UPPER_UNIT_TAG)
+				.replace(UPPER_UNIT_TAG, Renderer.count(KEY_COUNT_BASE).render(upperMoneyModel.getObject(), Locale.FRANCE));
+		LOGGER.warn("Expected: " + expected);
```

Comment:
It's no use logging the expected value, since assertEquals will display it if the assertion fails.